package.json

```diff
@@ -22,6 +22,7 @@
     "eslint": "^8.38.0",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.3.4",
+    "prettier": "2.8.8",
```

Comment:
you need to set up npm command to actually use it, like `"prettier:fix": "prettier 'src/**/*.{ts,tsx,json}' --write"`. Then make sure to run `npm run prettier:fix` automatically at some point, like in a Github Action pipeline.

Or, my recommendation, use a pre-commit git hook - https://typicode.github.io/husky/getting-started.html. You can use it for both eslint and prettier. It will check you code when you commit it locally, making sure no poop sneaks in.

We can set up a Github Action as a second line of defence later - just for practice.

Reply:
Also, I recommend installing Eslint and Prettier extensions for VS code. Eslint will show you the errors, Prettier can format code On save

Reply:
I set up ESLint and Prettier plugins in WebStorm. Will it suffice or I still have to do the pre-commit hook thing?

Reply:
Kind of, but there's a few points:
- you haven't yet formatted all the code - if I run `npx prettier --write .` it formats quite a few files.

For one-developer project it may be fine. But it's not reliable.
- if I commit to this repo and I don't have plugins installed, poorly formatted code will make it in to the code base
- if you get a new laptop, set up editor and forget to install plugins - you won't notice any warnings/errors

Also, I think it's worth learning how teams work on codebases - you won't always work alone - and code-formatting + linting are important parts of it these days